### ISSUE TITLE
Unincluded code in src / test directories

### ISSUE THREAD

**Issue by oxinabox** [julia-vscode/StaticLint.jl#217]
Thoughts on adding something to the linter to warn about files that are inside `test` but that are not included via any path to `runtests.jl`?
This is also a problem for `src` but for test it is bigger problem since tests that are not actually included always look like they pass.
And if you have been writng tests than excuting them via highlight + run selection, you can follow good TDD of writing a failing test first and still forget to actually every link those tests into your `runtests.jl`

**Comment by davidanthoff** [comment]
I'm worried about false positives here, though... We in principle can't detect all include statements (folks use loops and concatenate strings for includes etc.), so if we made this a default lint pass, we would certainly get false positives in some cases.

I've been thinking for a while that we might need something like a `strict` mode for the linter. If you turn that on, you can't use certain dynamic Julia features, but get much better linting messages.

**Comment by ZacLN** [comment]
Yep I agree this would be problematic, and with the idea of a strict mode
